src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12764,15 +12764,18 @@ LambdaExpressionSyntax parseLambdaExpressionWorker()
                     this.IsInAsync = true;
                 }
 
-                TypeSyntax returnType;
+                TypeSyntax returnType = null;
                 var resetPoint = this.GetResetPoint();
                 try
                 {
-                    returnType = ParseReturnType();
-                    if (CurrentToken.Kind != SyntaxKind.OpenParenToken)
+                    if (!IsVarType())
```

Comment:
>if (!IsVarType()) [](http://example.com/codeflow?start=20&length=17)

Should we instead parse this as a type and report an error that "var" is not allowed as a return type of a lambda?

Reply:
Perhaps even in binding.